worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/AbstractPlayerActor.java

```diff
@@ -520,22 +531,72 @@ public Region getLargestRegion() {
 
     @Override
     public void setSelection(Region region) {
-        RegionSelector selector;
-        if (region instanceof ConvexPolyhedralRegion) {
-            selector = new ConvexPolyhedralRegionSelector((ConvexPolyhedralRegion) region);
-        } else if (region instanceof CylinderRegion) {
-            selector = new CylinderRegionSelector((CylinderRegion) region);
-        } else if (region instanceof Polygonal2DRegion) {
-            selector = new Polygonal2DRegionSelector((Polygonal2DRegion) region);
-        } else {
-            selector = new CuboidRegionSelector(null, region.getMinimumPoint(),
-                    region.getMaximumPoint()
-            );
-        }
+        RegionSelector selector = switch (region) {
+            case ConvexPolyhedralRegion blockVector3s -> new ConvexPolyhedralRegionSelector(blockVector3s);
+            case CylinderRegion blockVector3s -> new CylinderRegionSelector(blockVector3s);
+            case Polygonal2DRegion blockVector3s -> new Polygonal2DRegionSelector(blockVector3s);
+            default -> new CuboidRegionSelector(null, region.getMinimumPoint(), region.getMaximumPoint());
+        };
         selector.setWorld(region.getWorld());
 
         getSession().setRegionSelector(getWorld(), selector);
     }
+
+    @Override
+    public void loadClipboardFromDisk() {
+        if (!clipboardLoading.tryAcquire()) {
+            if (!Fawe.isMainThread()) {
+                try {
+                    clipboardLoading.acquire();
+                    clipboardLoading.release();
+                } catch (InterruptedException e) {
+                    LOGGER.error("Error waiting for clipboard-on-disk loading for player {}", getName(), e);
+                }
+            }
+            return;
+        }
+
+        File file = MainUtil.getFile(
+                Fawe.platform().getDirectory(),
+                Settings.settings().PATHS.CLIPBOARD + File.separator + getUniqueId() + ".bd"
+        );
+        try {
+            Future<?> fut = Fawe.instance().submitUUIDKeyQueuedTask(getUniqueId(), () -> {
```

Comment:
Is it really wanted to enqueue twice? `LocalSession#loadCLipboardFromDisk(File)` submits a task in the KeyQueuedExecutorService - or is the threading purely for acquiring / synchronizing on the lock?

Reply:
That's done on purpose as we want to ensure that we offload from the main thread in AbstractPlayerActor, but without creating a new thread, or using up a thread in an unrelated thread pool. Since we will end up resubmitting to the queue pool again (which results in immediately running the task) we don't lose anything, but only gain safety in offloading in two places